pkg/volume/host_path/host_path.go

```diff
@@ -308,3 +323,175 @@ func getVolumeSource(spec *volume.Spec) (*v1.HostPathVolumeSource, bool, error)
 
 	return nil, false, fmt.Errorf("Spec does not reference an HostPath volume type")
 }
+
+func NewHostPathType(pathType string) *v1.HostPathType {
+	hostPathType := new(v1.HostPathType)
+	*hostPathType = v1.HostPathType(pathType)
+	return hostPathType
+}
+
+func newHostPathTypeList(pathType ...string) []*v1.HostPathType {
+	typeList := []*v1.HostPathType{}
+	for _, ele := range pathType {
+		typeList = append(typeList, NewHostPathType(ele))
+	}
+
+	return typeList
+}
+
+type hostPathTypeChecker interface {
+	Exists() bool
+	IsFile() bool
+	MakeFile() error
+	IsDir() bool
+	MakeDir() error
+	IsBlock() bool
+	IsChar() bool
+	IsSocket() bool
+}
+
+func getType(fileInfo os.FileInfo) (v1.HostPathType, error) {
+	mode := fileInfo.Sys().(*syscall.Stat_t).Mode
+	switch mode & syscall.S_IFMT {
+	case syscall.S_IFSOCK:
+		return v1.HostPathSocket, nil
+	case syscall.S_IFBLK:
+		return v1.HostPathBlockDev, nil
+	case syscall.S_IFCHR:
+		return v1.HostPathCharDev, nil
+	}
+
+	return "", fmt.Errorf("only recongnize socket, block device and character device")
+}
+
+// For mock up usage
```

Comment:
This is an anti-pattern.  Instead, how about something like:


```
func checkType(path string, pathType *v1.HostPathType) error {
	ftc, err := newOSFileTypeChecker(path)
    if err != nil {
        return err
    }
    return checkTypeInternal(ftc, path, pathType)
}

func checkTypeInternal(ftc fileTypeChecker, path string, pathType *v1.HostPathType) error {
    switch *pathType {
    // ...
}
```

This lets real code run with an osFileTypeChecker, but you can unit test `checkTypeInternal` with a fake checker, to exahaustively cover the corner cases.

Also, the osFileTypeChecker should have a full unit test to make sure it is handling all cases.